### ISSUE TITLE
dependency-check lint rule, --fix needs to be run twice

### ISSUE THREAD

**Issue by r4b6i0** [nrwl/nx#27412]
### Current Behavior

lib-01 `package.json` file has a missing peerDependency (`@angular/core`) and one dependency too many (`@angular/animations`).
In order for both errors to be fixed I need to run `nx lint lib-01 --fix` two consecutive times

### Expected Behavior

Both errors in lib-01 `package.json` file should be fixed by the first `nx lint lib-01 --fix` run

### GitHub Repo

https://github.com/r4b6i0/nx-dependency-checks-bug

### Steps to Reproduce

1. check that lib-01 `package.json` file contains the peerDependencies `@angular/common` and `@angular/animations`, which are not correct based on lib-01 `import` statements
2. run the command `nx lint lib-01 --fix`
3. `package.json` file now contains only the peerDependency `@angular/common`
4. run the command `nx lint lib-01 --fix` again
5. `package.json` file now contains both the peerDependencies `@angular/common` and `@angular/core` which are the correct dependencies based on lib-01 `import` statements

### Nx Report

```shell
Node   : 18.20.4
OS     : win32-x64
npm    : 10.7.0

nx                 : 18.3.5
@nx/js             : 18.3.5
@nx/jest           : 18.3.5
@nx/linter         : 18.3.5
@nx/eslint         : 18.3.5
@nx/workspace      : 18.3.5
@nx/angular        : 18.3.5
@nx/cypress        : 18.3.5
@nx/devkit         : 18.3.5
@nx/eslint-plugin  : 18.3.5
@nrwl/tao          : 18.3.5
@nx/web            : 18.3.5
@nx/webpack        : 18.3.5
typescript         : 5.4.5
---------------------------------------
Registered Plugins:
@nx/cypress/plugin
@nx/eslint/plugin
```

### Package Manager Version

npm 10.7.0

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

### Additional Information

To create this example I started with a new empty nx@18.3.5 repo using the command `npx create-nx-workspace@18.3.5`, added a new angular library with the command `nx g @nx/angular:library lib-01 --publishable --import-path=@my-org/my-lib-01`, modified the library's `package.json` so that the `peerDependencies` are not correct.
I did nothing else.

**Comment by r4b6i0** [comment]
Thanks @leosvelperez. Do you think this fix will be released also for branch `18.3.x`?

**Comment by r4b6i0** [comment]
@FrozenPandaz do you know if the fix for #27476 will be released also for branch `18.3.x`?
